packages/@aws-cdk/aws-eks/README.md

```diff
@@ -464,6 +464,19 @@ Kubernetes secrets using the AWS Key Management Service (AWS KMS) can be enabled
 on [creating a cluster](https://docs.aws.amazon.com/eks/latest/userguide/create-cluster.html)
 can provide more details about the customer master key (CMK) that can be used for the encryption.
 
+By default, `eks.Cluster` is created using AWS-managed encryption keys to protect the `etcd` volumes at disk-level. To enable envelope encryption feature to encrypt/decrypt sensitive data stored in Kubernetes secrets, you can supply your own external KMS key by specifying the `encryptionKey` props:
+
+```ts
+import * as kms from "@aws-cdk/aws-kms";
+
+const key = new kms.Key(this, 'Key');
+
+new eks.Cluster(this, 'Cluster', {
+  version: eks.KubernetesVersion.V1_16,
+  encryptionKey: key,
```

Comment:
Since this only encrypts secrets, I wonder if the name should be `secretsEncryptionKey`?

Reply:
totally agreed, it provides much clear context of how does this KMS CMK applies to without referring to the documentation. I will follow your advise and submit another round of PR. Thanks @eladb !